src/components/Table/components/TableToolbar/components/ColumnGrouping/columnGrouping.tsx

```diff
@@ -0,0 +1,53 @@
+import { ListItemText } from "@mui/material";
+import { RowData } from "@tanstack/react-table";
+import React from "react";
+import { DropdownButton } from "../../../../../common/Button/components/DropdownButton/dropdownButton";
+import { DropdownMenu } from "../../../../../common/DropdownMenu/dropdownMenu";
+import { handleClearGroupingState } from "../../../TableFeatures/ColumnGrouping/utils";
+import { StyledMenuItem } from "./columnGrouping.styles";
+import { MenuItems } from "./components/MenuItems/menuItems";
+import { LIST_ITEM_TEXT_PROPS, MENU_ITEM_PROPS, MENU_PROPS } from "./constants";
+import { ColumnGroupingProps } from "./types";
+import { getButtonLabel, getColumnGrouping } from "./utils";
+
+export const ColumnGrouping = <T extends RowData>({
+  tableInstance,
+}: ColumnGroupingProps<T>): JSX.Element | null => {
+  const {
+    getState,
+    options: { enableGrouping },
+  } = tableInstance;
+  if (!enableGrouping) return null;
+  const { grouping } = getState();
+  const groupingByColumnId = getColumnGrouping(tableInstance);
+  return (
+    <DropdownMenu
+      {...MENU_PROPS}
+      button={(props) => (
+        <DropdownButton {...props}>
+          {getButtonLabel(groupingByColumnId, grouping)}
```

Comment:
DropdownButton label is updated, depending on grouping state. This update, has meant we needed to update how we pass in the button for DropdownMenu.